portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -142,4 +147,26 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.delete(commentEntityKey);
   }
+
+  /**
+   * Method that returns parameter text's sentiment score
+   * Returns a value in [-1, 1], representing how negative or positive text is
+   */
+  private static float getSentimentScore(String text) throws IOException {
+      // Init doc element containing text
+      Document doc =
+        Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
+
+      // Init the languageService
+      LanguageServiceClient languageService = LanguageServiceClient.create();
+    
+      // Analyze the doc's sentiment using the languageService and get a score
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+      float score = sentiment.getScore();
```

Comment:
I am currently getting the sentiment score as a float (this is how the API returns it), saving it in datastore, and retrieving it as a Double (couldn’t find datastore support for floats). Is there a better approach for this? 
You can find the code here -> refer to lines 87 and 118

Reply:
What if you cast it to Double here and make everything deal with the score as a double? There is no loss of precision when converting float to double.

Reply:
Also, I think score must be recalculated every time the page is loaded? 
This is so, if the API changes its behavior it is reflected here as well.

Reply:
> What if you cast it to Double here and make everything deal with the score as a double? There is no loss of precision when converting float to double.

Thanks for suggestion! I wanted to do that before, but I found out there is no straightforward way to convert from a float to a Double (you have to go through a string). Popular ways of converting are [here](http://programmingjungle.blogspot.com/2013/03/float-to-double-conversion-in-java.html#:~:text=There%20are%20a%20three%20popular,(new%20Float(fvar).). FloatingDecimal class is no longer supported (can't import it) and the others look a little ugly to me, do you think I should do it like this?

Reply:
> Also, I think score must be recalculated every time the page is loaded?
> This is so, if the API changes its behavior it is reflected here as well.

That's a good point, thank you! I think it makes more sense to do that, but the [walkthrough](https://github.com/PredaMihaiDragos/Google-STEP/blob/master/walkthroughs/week-4-libraries/sentiment-analysis/sentiment-analysis-walkthrough.md) says to calculate it when the user enters it and to store it in the database and I just followed it... Do you think I should change the logic?

Reply:
> > What if you cast it to Double here and make everything deal with the score as a double? There is no loss of precision when converting float to double.
> 
> Thanks for suggestion! I wanted to do that before, but I found out there is no straightforward way to convert from a float to a Double (you have to go through a string). Popular ways of converting are [here](http://programmingjungle.blogspot.com/2013/03/float-to-double-conversion-in-java.html#:~:text=There%20are%20a%20three%20popular,(new%20Float(fvar).) FloatingDecimal class is no longer supported (can't import it) and the others look a little ugly to me, do you think I should do it like this? In my opinion letting the datastore class cast it looks cleaner

I think it is fine the way it is implemented right now.

Reply:
> Do you think I should change the logic?

I think it's fine this way. But it's something you can keep in mind for future while storing results of API calls.